### ISSUE TITLE
Rule Change: [id-match] `allows` option

### ISSUE THREAD

**Issue by RedGuy12** [eslint/eslint#16728]
### What rule do you want to change?

id-match

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/*eslint id-match: ["error", "^[a-z]+([A-Z][a-z]+)*$"]*/

var my_favorite_color = "#112C85";
```

### What does the rule currently do for this code?

Currently, this rule would error on `my_favorite_color`.

### What will the rule do after it's changed?

I would like an option to ignore specific identifiers. This code would not error:
```js
/*eslint id-match: ["error", "^[a-z]+([A-Z][a-z]+)*$", { ignores: ["my_favorite_color"] }]*/

var my_favorite_color = "#112C85";
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @RedGuy12, thanks for the issue!

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we are not adding new options to the `id-match` rule.

You can add allowed identifiers to the regular expression to get the same functionality:

```js
/*eslint id-match: ["error", "^(?:[a-z]+([A-Z][a-z]+)*|my_favorite_color)$"]*/

var my_favorite_color = "#112C85";

```

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbWF0Y2g6IFtcImVycm9yXCIsIFwiXig/OlthLXpdKyhbQS1aXVthLXpdKykqfG15X2Zhdm9yaXRlX2NvbG9yKSRcIl0qL1xuXG52YXIgbXlfZmF2b3JpdGVfY29sb3IgPSBcIiMxMTJDODVcIjtcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiZXM2Ijp0cnVlfX19)
